core/src/avm1/globals/movie_clip.rs

```diff
@@ -761,86 +761,90 @@ fn duplicate_movie_clip<'gc>(
     activation: &mut Activation<'_, 'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    // duplicateMovieClip method uses biased depth compared to CloneSprite
-    duplicate_movie_clip_with_bias(movie_clip, activation, args, AVM_DEPTH_BIAS)
-}
-
-pub fn duplicate_movie_clip_with_bias<'gc>(
-    movie_clip: MovieClip<'gc>,
-    activation: &mut Activation<'_, 'gc>,
-    args: &[Value<'gc>],
-    depth_bias: i32,
-) -> Result<Value<'gc>, Error<'gc>> {
-    let (new_instance_name, depth) = match &args.get(0..2) {
-        Some([new_instance_name, depth]) => (
-            new_instance_name.coerce_to_string(activation)?,
-            depth.coerce_to_i32(activation)?.wrapping_add(depth_bias),
-        ),
-        _ => {
+    let name = match args.get(0) {
+        Some(name) => name.coerce_to_string(activation)?,
+        None => {
             avm_error!(
                 activation,
                 "MovieClip.duplicateMovieClip: Too few parameters"
             );
             return Ok(Value::Undefined);
         }
     };
-    let init_object = args.get(2);
+    let depth = match args.get(1) {
+        Some(depth) => depth.coerce_to_i32(activation)?,
+        None => 0,
+    };
+    let init_object = args.get(2).map(|v| v.coerce_to_object(activation));
```

Comment:
Nit: Maybe it could be worth to add a comment mentioning that this argument (`initObject`) is not version-gated. The docs say it is supported only in FP6 and later, but this doesn't appear to be true. I suppose this should refer to the player version and not the SWF version, but the documentation also uses the same wording when talking about the return value (and you found out it was indeed version-gated).

https://open-flash.github.io/mirrors/as2-language-reference/MovieClip.html#duplicateMovieClip()